src/types/mod.rs

```diff
@@ -257,12 +257,23 @@ impl ValueInternal for JsString {
 }
 
 impl JsString {
+    #[cfg(feature = "legacy-runtime")]
     pub fn size(self) -> isize {
         unsafe {
             neon_runtime::string::utf8_len(self.to_raw())
         }
     }
 
+    #[cfg(feature = "napi-runtime")]
+    pub fn size<'a, C: Context<'a>>(self, cx: &mut C) -> isize {
+        let env = cx.env().to_raw();
+
+        unsafe {
+            neon_runtime::string::utf8_len(env, self.to_raw())
+        }
+    }
+
+    #[cfg(feature = "legacy-runtime")]
     pub fn value(self) -> String {
```

Comment:
would it make sense to also add a dummy Context argument here? then the tests and stuff could be run with either feature. Or is the idea to fully maintain back compat in the legacy runtime?

Reply:
The idea is to not make any breaking changes where we aren't adding features. We would like to keep all of the N-API work on the master branch, but don't want to introduce upgrade churn for users that aren't taking advantage.

Once N-API is complete, I would like to revisit some of these, but I think it makes sense for now to add technical debt to keep legacy APIs consistent.